### PR TITLE
Feature#9/crud companies

### PR DESCRIPTION
**Pull request by claunicole**
# Description
-Create the CRUD of the company's branches
-Create new, edit and show branches views
-Create the form for create and edit a branch

 # Testing
-Run rails server, go to /companies/new to create a new branch
-In the /companies route you can edit and delete a company branch

  ## Screenshots
company-index

### CODE REVIEW

**Comment by JorgeRodz** [comment]
The code looks good but I will rather use another color palette.

**Review by JorgeRodz — APPROVED** [review]

File: app/controllers/companies_controller.rb [hunk 1]
```diff
@@ -2,6 +2,48 @@
...
+    status: :see_other
+  end
+
+  private
+  def company_params
+    params.require(:company).permit(:name,:address)
   end
 end
+
+#def destroy @article = Article.find(params[:id]) @article.destroy redirect_to root_path, status: :see_other end private def article_params params.require(:article).permit(:title, :body) end
```
Comment (willcab) [inline app/controllers/companies_controller.rb]: just delete the commented code, the rest is fine.

File: config/routes.rb [hunk 2]
```diff
@@ -3,8 +3,14 @@
 Rails.application.routes.draw do
   get 'companies', to: 'companies#index'
   get 'employees', to: 'employees#index'
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get 'companies/new'
+  get 'companies/:id', to: 'companies#show', as:'show_company'
+  get 'companies/:id/edit', to: 'companies#edit', as:'edit_company'
+  post 'companies', to: 'companies#create'
+  delete 'companies/:id', to: 'companies#destroy', as:'delete_company'
+  patch 'companies/:id', to: 'companies#update', as:'company'
```
Comment (HopeSaurus) [inline config/routes.rb]: Have in mind to use "resources: companies"  that creates all of these lines.

**Review by willcab — APPROVED** [review]
LGTM 👍

**Review by HopeSaurus — APPROVED** [review]
Its all good, just a quick heads up.

File: app/views/companies/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        <%= f.text_field :address, class:"form-title" %>               
+      </div>
+    </div>            
+    <div class="form-group">
+      <div class="col-md-12 text-center" >
+        <%= f.submit "new", class:"button_header button_new"  %>
+      </div>
+    </div>  
+  </div>
+<%end%>
```
Comment (JuanequeX) [inline app/views/companies/edit.html.erb]: Add a blank line here

File: app/views/companies/index.html.erb [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+  <table class="table">
+    <thead>
+      <tr>
+      <th scope="row" >Name</th>
+      <th scope="row" >Address</th>
+      <th scope="row" >Id</th> 
+      <th scope="row" ></th> 
+      <th scope="row" ></th>
+      
+
```
Comment (JuanequeX) [inline app/views/companies/index.html.erb]: delete this two extra spaces

File: app/views/companies/index.html.erb [hunk 5]
```diff
@@ -1,5 +1,42 @@
...
+<%# <table id="myTable" class="table"> 
+<thead> 
+<tr> 
+
+<th scope="col" class="text-center">Actual</th> 
+<th scope="col" class="text-center">Media</th> 
+<th scope="col" class="text-center">% Dif</th> 
+</tr> 
+</thead>  %>
+
```
Comment (JuanequeX) [inline app/views/companies/index.html.erb]: Delete this innecessary code

File: app/views/companies/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+          <%= f.text_field :address, class:"form-title" %>               
+        </div>
+    </div>            
+    <div class="form-group">
+      <div class="col-md-12 text-center" >
+        <%= f.submit "new", class:"button_header" %>
+      </div>
+    </div>  
+  </div>
+<%end%>
```
Comment (JuanequeX) [inline app/views/companies/new.html.erb]: Add a blank line here

File: app/views/companies/show.html.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+<h2 class="title"><%= @companies.name %></h2>
```
Comment (JuanequeX) [inline app/views/companies/show.html.erb]: add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/controllers/companies_controller.rb (hunk 1 above)
Comment (claunicole) [reply app/controllers/companies_controller.rb]: Done

File: app/views/companies/show.html.erb (hunk 7 above)
Comment (claunicole) [reply app/views/companies/show.html.erb]: Done

File: app/views/companies/new.html.erb (hunk 6 above)
Comment (claunicole) [reply app/views/companies/new.html.erb]: Done

**Review by JuanequeX — APPROVED** [review]
LGTM

**Pull request merged**